src/test/java/uk/nhs/tis/trainee/notifications/service/NotificationServiceTest.java

```diff
@@ -518,6 +591,23 @@ void shouldMapUserDetailsWhenCognitoAccountNotExist() {
     assertThat("Unexpected gmc number", expectedResult.gmcNumber(), is(USER_GMC));
   }
 
+  @Test
```

Comment:
You could tweak these two tests to also demonstrate that the gmc is trimmed, if you set them up as something like
```
@ParameterizedTest
@ValueSource(strings = "  some gmc number  ")
@NullSource
void shouldMapUserDetailsWhenCognitoAccountNotExist(String gmc) {
...
if (gmc == null) {
      assertThat("Unexpected gmc number.", result.gmcNumber(), is(nullValue()));
    } else {
      assertThat("Unexpected gmc number.", result.gmcNumber(), is(gmc.trim()));
    }
}
```